File: 776. Split BST/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def splitBST(self, root: TreeNode, V: int) -> List[TreeNode]:
        if not root:
            return None, None
        
        if root.val <= V:
            low, high = self.splitBST(root.right, V)
            root.right = low
            return root, high
        else:
            low, high = self.splitBST(root.left, V)
            root.left = high
            return low, root

File: 776. Split BST/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public TreeNode[] splitBST(TreeNode root, int V) {
        if (root == null){
            return new TreeNode[] {null, null};
        }
        
        if (root.val <= V) {
            TreeNode[] right = splitBST(root.right, V);
            root.right = right[0];
            right[0] = root;
            return right;
        } else {
            TreeNode[] left = splitBST(root.left, V);
            root.left = left[1];
            left[1] = root;
            return left;
        }
    }
}
